pkg/rtc/participant.go

```diff
@@ -78,7 +78,7 @@ type ParticipantImpl struct {
 	resSink             atomic.Value // routing.MessageSink
 	resSinkValid        atomic.Bool
 	subscriberAsPrimary bool
-	grants              atomic.Value // *auth.ClaimGrants
+	grants              *auth.ClaimGrants
```

Comment:
How about copy on write , since we only have 2 methods modify the grants (SetMetaData and SetPermission), can add a clone method to ClaimGrants, and copy it when modify.

Reply:
We can do that. Is there a standard way to implement deep copy in Golang? If we do `copy-on-write`, will look at the changes again here and modify. If there is a `Clone` method, I am fine keeping the current changes and doing a clone while returning grants. What do you think?

Reply:
Unfortunately there is no standard way to deep copy a struct... we need do it manually